### PR TITLE
Aheals now fix addictions and sanity.

### LINKED ISSUE #43578: Admin heal does not clear addictions.
## Reproduction:

What it says on the tin. This was fixed a long time ago but seems to have broken again.
See #8894

Note that I had the smoker trait, so it's reasonable that my smoking addiction wouldn't be cleared, but meth absolutely should have been.

### LINKED ISSUE #43589: Admin heal does not restore sanity.
## Round ID:
105987

## Reproduction:
1. Set sanity to very low or very high (found in add_mood in VV)
2. Admin heal
3. Sanity is not touched.

It should probably restore it to 100.
A little bit strange because it DOES clear moods, but it doesn't restore sanity.

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

See title.
Fixes #43578
Fixes #43589

## Why It's Good For The Game

Aheals should fix everything and they don't, that's sad. Also buffs implanted regen cores and slime extracts, but that's really whatever.

## Changelog
:cl:
fix: Aheals now remove addictions and restore your mood to default.
/:cl:

### CODE REVIEW

**Comment by bgobandit** [comment]
it is kind of questionable that in-game items use admin healing

**Comment by nemvar** [comment]
Yes, but that's a seperate issue.

File: code/datums/components/mood.dm [hunk 1]
```diff
@@ -356,5 +357,10 @@
 	else
 		clear_event(null, "area")
 
+/datum/component/mood/proc/on_revive(datum/source, full_heal)
+	if(!full_heal)
+		return
+	setSanity(initial(sanity))
```
Comment (Qustinnus) [inline code/datums/components/mood.dm]: move remove temp moodies proc here instead of where it was before

Comment (nemvar) [reply code/datums/components/mood.dm]: done

File: code/datums/components/mood.dm [hunk 2]
```diff
@@ -280,14 +282,14 @@
...
-/datum/component/mood/proc/remove_temp_moods(var/admin) //Removes all temp moods
+/datum/component/mood/proc/remove_temp_moods() //Removes all temp moods
 	for(var/i in mood_events)
 		var/datum/mood_event/moodlet = mood_events[i]
 		if(!moodlet || !moodlet.timeout)
 			continue
 		mood_events -= moodlet.category
 		qdel(moodlet)
-		update_mood()
+	update_mood()
```
Comment (nemvar) [inline code/datums/components/mood.dm]: This is mostly a change to improve how well this proc runs. We don't have anything that would require updating mood for every single removed moodlet. At least not currently.

**Comment by Dennok** [comment]
> it is kind of questionable that in-game items use admin healing

There is admins use ingame healing by button.

File: code/datums/components/mood.dm (hunk 2 above)
Comment (Qustinnus) [reply code/datums/components/mood.dm]: good god how did i miss this, thanks.

**Pull request merged**